src/main/kotlin/calculator/Calculator.kt

```diff
@@ -0,0 +1,77 @@
+package calculator
+
+private const val DELIMITER = " "
+
+/**
+ * 문자열 계산기
+ * */
+object Calculator {
+
+    enum class CalculatorSign(val sign: String) {
+        PLUS("+") {
+            override fun calculate(num1: Int, num2: Int): Int = num1 + num2
+        },
+        MINUS("-") {
+            override fun calculate(num1: Int, num2: Int): Int = num1 - num2
+        },
+        MULTIPLY("*") {
+            override fun calculate(num1: Int, num2: Int): Int = num1 * num2
+        },
+        DIVIDE("/") {
+            override fun calculate(num1: Int, num2: Int): Int = num1 / num2
+        };
+
+        /**
+         * 숫자 두개와 부호로 계산한 값을 반환
+         *
+         * @param num1 입력 값1
+         * @param num2 입력 값2
+         * @return 계산된 값
+         * */
+        abstract fun calculate(num1: Int, num2: Int): Int
+    }
+
+    private fun checkError(splitFormulaList: List<String>) {
+
+        require(splitFormulaList.isNotEmpty()) {
+            "Invalid Expression: $splitFormulaList"
+        }
+
+        require(splitFormulaList.all { it.isNotBlank() }) {
+            "Invalid Expression: $splitFormulaList"
+        }
+
+        require(isCheckIndexValue(splitFormulaList)) {
+            "Invalid Expression: $splitFormulaList"
+        }
+    }
+
+    private fun isCheckIndexValue(inputStrSplitList: List<String>): Boolean =
+        inputStrSplitList.mapIndexed { index, value ->
+            if (index % 2 == 0) {
+                value.toIntOrNull() != null
+            } else {
+                CalculatorSign.values().any { sign -> sign.sign == value }
+            }
+        }.all { it }
+
+    /**
+     * 계산 요청시 사용
+     *
+     * @param inputStr 계산에 사용할 문자열
+     * @return 계산식을 이용해 얻은 결과 값
+     * */
+    fun calculate(inputStr: String?): Int {
+
+        val splitFormulaList: List<String> = inputStr?.split(DELIMITER) ?: emptyList()
+
+        checkError(splitFormulaList)
+
+        val splitFormulaPair = splitFormulaList.partition { it.toIntOrNull() != null }
```

Comment:
```suggestion
        splitFormulaList.foldIndexed(ArrayDeque<String>() to ArrayDeque<String>()) { index, (operators, operands), item ->
            if (index % 2 == 0) {
                operators.add(item)
            } else {
                operands.add(item)
            }
            operators to operands
        }
```

이런식으로도 할 수 있어요 😀 참고만 해보세용 ㅎㅎ